docs/writing-a-plugin.md

```diff
@@ -241,72 +246,32 @@ We'll use both Vanilla JavaScript & ES6 examples.
 
 1 . In **_my-plugin.js_**, define the plugin class:
 
-### _Plain JavaScript_
-
-```js
-var MyPlugin = function (name, player, config) {};
-```
-
-### _ES6_
-
 ```typescript
 class MyPlugin {
   constructor(name, player, config) {}
 }
 ```
 
-2 . Next, and add the code that will inherit from `BasePlugin`:
-
-### _Plain JavaScript_
-
-```js
-var MyPlugin = function (name, player, config) {
-  KalturaPlayer.core.BasePlugin.call(this, name, player, config);
-};
-
-MyPlugin.createPlugin = KalturaPlayer.core.BasePlugin.createPlugin;
-
-MyPlugin.prototype = new KalturaPlayer.core.BasePlugin();
-```
-
-### _ES6_
+2 . Extends the `BasePlugin` base class:
 
 ```typescript
-class MyPlugin extends KalturaPlayer.core.BasePlugin {
+import {BasePlugin} from 'kaltura-player-js';
+
+class MyPlugin extends BasePlugin {
   constructor(name, player, config) {
     super(name, player, config);
   }
 }
 ```
 
-## Step 3 - Override the `BasePlugin` Methods
+## Step 3 - Implement the `BasePlugin` Methods
 
 In **_my-plugin.js_**, add the necessary methods and properties to override as explained above:
 
-### _Plain JavaScript_
-
-```js
-var MyPlugin = function (name, player, config) {
-  KalturaPlayer.core.BasePlugin.call(this, name, player, config);
-};
-
-MyPlugin.createPlugin = KalturaPlayer.core.BasePlugin.createPlugin;
-
-MyPlugin.prototype = new KalturaPlayer.core.BasePlugin();
-
-MyPlugin.defaultConfig = {};
-
-MyPlugin.isValid = function (player) {};
-
-MyPlugin.prototype.destroy = function () {};
-
-MyPlugin.prototype.reset = function () {};
-```
-
-### _ES6_
-
 ```typescript
-class MyPlugin extends KalturaPlayer.core.BasePlugin {
+import {BasePlugin} from 'kaltura-player-js';
```

Comment:
I would use the original.
If you use import syntax you also need to explain how we use extern in webpack config to link to the specific player/preact instance, but the customer might not be using webpack at all.

Reply:
you don’t need Webpack in order to use `import` in ES6 you just need set the script to `type=module` as i did

Reply:
but you want to import from the local player, right ?